onnxruntime/test/onnx/main.cc

```diff
@@ -286,147 +287,160 @@ int real_main(int argc, char* argv[], Ort::Env& env) {
+  broken_tests.insert("keras2coreml_linear_sklearn_load_breast_cancer", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_linear_ImageNet_small", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_linear_ImageNet_large", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_linear_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_leakyrelu_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_hard_sigmoid_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_elu_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_Dense_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_Conv2D_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("coreml_VGG16_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("coreml_Resnet50_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("coreml_Inceptionv3_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("coreml_FNS-Candy_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("coreml_AgeNet_ImageNet", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_thresholdedrelu_ImageNet_large", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_thresholdedrelu_ImageNet_small", {}, "This model uses contrib ops."};
+  broken_tests.insert("keras2coreml_thresholdedrelu_sklearn_load_breast_cancer", {}, "This model uses contrib ops."};
+  broken_tests.insert("thresholdedrelu", {}, "This model uses contrib ops."};
+  broken_tests.insert("thresholdedrelu_default", {}, "This model uses contrib ops."};
+  broken_tests.insert("dynamic_slice_default_axes", {}, "This model uses contrib ops."};
+  broken_tests.insert("thresholdedrelu_example", {}, "This model uses contrib ops."};
+  broken_tests.insert("dynamic_slice_neg failed", {}, "This model uses contrib ops."};
+  broken_tests.insert("dynamic_slice_start_out_of_bounds", {}, "This model uses contrib ops."};
+  broken_tests.insert("dynamic_slice", {}, "This model uses contrib ops."};
+  broken_tests.insert("dynamic_slice_end_out_of_bounds", {}, "This model uses contrib ops."};
+  broken_tests.insert("dynamic_slice_neg", {}, "This model uses contrib ops."};
 #endif
 
   int result = 0;
-  for (const std::string& s : stat.GetFailedTest()) {
-    if (broken_tests.find(s) == broken_tests.end()) {
-      fprintf(stderr, "test %s failed, please fix it\n", s.c_str());
+  for (const auto& p: stat.GetFailedTest()) {
+    BrokenTest t{p.first, {}, ""};
+    auto iter = broken_tests.find(t);
+    if (iter == broken_tests.end() ||
+       (p.second != -1 && !iter->broken_versions_.empty() && iter->broken_versions_.find(p.second) == iter->broken_versions_.end())) {
+      fprintf(stderr, "test %s failed, please fix it\n", p.first.c_str());
```

Comment:
>test %s failed, please fix it [](start = 23, length = 29)

Can we not run it since we can detect this condition before running?

Reply:
Yes we can, but it requires a bit of architectural changes - broken_tests is not getting passed down into test run stack. Some further thinking is why not just letting it run, as long as it is not blocking the test pipeline it might be helpful to see which case, expectedly, are failing the test.